### ISSUE TITLE
Add is_update to beforeSave Hook as well

### ISSUE THREAD

**Issue by PhilippGrashoff** [atk4/data#456]
Hi, some month ago the afterSave hook got the very sensible second parameter is_update. beforeSave does not have this. It could be done very easily switching some lines:
Current:
```
            if ($this->hook('beforeSave') === false) {
                return $this;
            }

            $is_update = $this->loaded();
```

Switched:
```
            $is_update = $this->loaded();

            if ($this->hook('beforeSave', $is_update) === false) {
                return $this;
            }
```

Of course its not as important as in the afterSave hook, as you can easily check inside the beforeSave hook if the model is loaded or not, but that would make both hooks behave the same way.

**Comment by abbadon1334** [comment]
LGTM even if we already have beforeInsert and beforeUpdate to clearly separate the operation.
This can shortens by 2 or 3 line the code.

**Comment by romaninsh** [comment]
I'm all for consistency, so please go ahead and PR. No need to discuss even.

**Comment by DarkSide666** [comment]
Yeah, looks very good. Can you make PR?

**Comment by PhilippGrashoff** [comment]
Will do

**Comment by DarkSide666** [comment]
implemented in #462
